### PR TITLE
detect: Add missing keyword URLs and description

### PR DESCRIPTION
**Pull request by shivan1b**
Add missing keyword URLs and their description. Fix the ones that
were incorrect.

Additionally, tickets corresponding to all the keywords whose documentations are not available have been created.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/2974

### CODE REVIEW

File: src/detect-replace.c [hunk 1]
```diff
@@ -69,6 +69,8 @@ static int DetectReplacePostMatch(ThreadVars *tv,
 void DetectReplaceRegister (void)
 {
     sigmatch_table[DETECT_REPLACE].name = "replace";
+    sigmatch_table[DETECT_REPLACE].desc = "only be used in ips. changes the content it follows into another";
```
Comment (victorjulien) [inline src/detect-replace.c]: s/in ips/in IPS-mode/g.
Replaces packet content with the value in this keyword. Or something.

File: src/detect-uricontent.c [hunk 2]
```diff
@@ -68,6 +68,8 @@ static int g_http_uri_buffer_id = 0;
 void DetectUricontentRegister (void)
 {
     sigmatch_table[DETECT_URICONTENT].name = "uricontent";
+    sigmatch_table[DETECT_URICONTENT].desc = "match specifically and only on the request URI buffer";
```
Comment (victorjulien) [inline src/detect-uricontent.c]: legacy keyword to match on the request URI buffer

File: src/detect-hostbits.c [hunk 3]
```diff
@@ -78,7 +78,7 @@ void DetectHostbitsRegister (void)
 {
     sigmatch_table[DETECT_HOSTBITS].name = "hostbits";
     sigmatch_table[DETECT_HOSTBITS].desc = "operate on host flag";
-//    sigmatch_table[DETECT_HOSTBITS].url = DOC_URL DOC_VERSION "/rules/flow-keywords.html#flowbits";
+    sigmatch_table[DETECT_HOSTBITS].url = DOC_URL DOC_VERSION "/rules/xbits.html#unix-socket";
```
Comment (victorjulien) [inline src/detect-hostbits.c]: this link is not very accurate. We're supposed to point to rule syntax and such, and this is more about unix-socket interaction

File: src/detect-geoip.c [hunk 4]
```diff
@@ -52,6 +52,8 @@ static int DetectGeoipSetupNoSupport (DetectEngineCtx *a, Signature *b, const ch
 void DetectGeoipRegister(void)
 {
     sigmatch_table[DETECT_GEOIP].name = "geoip";
+    sigmatch_table[DETECT_GEOIP].desc = "enables (you) to match on the source, destination or source and destination IP addresses of network traffic, and to see to which country it belongs.";
```
Comment (victorjulien) [inline src/detect-geoip.c]: the 'enables (you)' part is a bit of a different style. What about just 'match on the source...'?

**Comment by shivan1b** [comment]
Replaced by #3984

**Pull request closed without merging**